src/fava/core/watcher.py

```diff
@@ -18,28 +17,45 @@
 if TYPE_CHECKING:  # pragma: no cover
     import types
 
+    from watchfiles.main import Change
+
 
 log = logging.getLogger(__name__)
 
 
 class _WatchfilesThread(threading.Thread):
-    def __init__(self, paths: list[Path], mtime: int) -> None:
+    def __init__(
+        self, files: list[Path], folders: list[Path], mtime: int
+    ) -> None:
         super().__init__(daemon=True)
         self._stop_event = threading.Event()
-        self.paths = paths
         self.mtime = mtime
+        self._file_set = {f.absolute() for f in files}
+        self._folder_set = {f.absolute() for f in folders}
 
     def stop(self) -> None:
         """Set the stop event for watchfiles and join the thread."""
         self._stop_event.set()
         self.join()
 
+    def _is_relevant(self, _c: Change, path: str) -> bool:
+        return Path(path) in self._file_set or any(
+            Path(path).is_relative_to(d) for d in self._folder_set
+        )
+
     def run(self) -> None:
         """Watch for changes."""
         atexit.register(self.stop)
 
-        for changes in watch(*self.paths, stop_event=self._stop_event):
-            for _, path in changes:
+        watch_paths = {f.parent for f in self._file_set} | self._folder_set
+
+        for changes in watch(
+            *watch_paths,
+            stop_event=self._stop_event,
+            ignore_permission_denied=True,
```

Comment:
This causes the "old_deps" builds to fail. It's here for https://github.com/samuelcolvin/watchfiles/pull/224 ; it's only necessary if you end up watching a directory that contains unreadable files.

Reply:
That's a legimitate failure - we either need to bump the required version of watchfiles or write this in a way such that it doesn't fail with older versions. I guess bumping the version would be fine and easier.

Reply:
It's probably not a dealbreaker: this is only necessary in the case that the directory with the Beancount file also contains files you cannot read; I hit this testing with `/tmp`, and added this in response.

However, this only takes the minimum version from [`0.19.0`](https://github.com/samuelcolvin/watchfiles/releases/tag/v0.19.0) to [`0.20.0`](https://github.com/samuelcolvin/watchfiles/releases/tag/v0.20.0), which seems reasonable so I've bumped that in the PR.